pkg/cortex/cortex.go

```diff
@@ -238,7 +241,22 @@ func (t *Cortex) initModule(cfg *Config, m moduleName) error {
 
 // Run starts Cortex running, and blocks until a signal is received.
 func (t *Cortex) Run() error {
-	return t.server.Run()
+	var err error
+	done := make(chan struct{})
+	go func() {
+		err = t.server.Run()
+
+		close(done)
+	}()
+
+	// Initiate shutdown if its a job to flush data.
+	if t.cfg.Ingester.IsFlushJob {
+		_ = t.Stop()
```

Comment:
We need to make sure that calling `t.Stop()` twice is not a problem. Currently, `main.go` would also call this and try stopping the modules again.